core/src/lib.rs

```diff
@@ -3,5 +3,120 @@ pub mod error;
 
 pub mod serialization;
 
+pub mod metadata;
+
 pub mod ordered_hash_map;
 
+use crate::serialization::{
+  Serialize,
+  Deserialize,
+  StringEncoding,
+  fit_sz,
+};
+
+extern crate derivative;
+use derivative::Derivative;
+
+use std::io::{Write, BufRead, Seek};
+use cbor_event::{se::Serializer, de::Deserializer};
+
+pub type Epoch = u64;
+
+pub type Slot = u64;
+
+pub type TransactionIndex = u16;
+
+pub type CertificateIndex = u64;
+
+#[derive(Clone, Debug, serde::Deserialize, serde::Serialize, schemars::JsonSchema, Derivative)]
+#[derivative(Eq, PartialEq, Ord="feature_allow_slow_enum", PartialOrd="feature_allow_slow_enum", Hash)]
```

Comment:
pulled from `chain/src/lib.rs` since it's used in other places like metadata and seems general enough (supporting CDDL's `int` type)